numpy/lib/tests/test_function_base.py

```diff
@@ -1499,6 +1500,59 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_frompyfunc_leaks(self):
+        # exposed in gh-11867 as np.vectorized, but the problem stems from
+        # frompyfunc.
+        # class.attribute = np.frompyfunc(<method>) creates a
+        # reference cycle that requires a gc collection cycle to break
+        # (on CPython 3)
+        import gc
+        gc.disable()
+        iters = 20
+        try:
+            class A(object):
+                def method(self, *args):
+                    return 0
+
+                @classmethod
+                def method1(*args):
```

Comment:
nit: classmethods should have a `cls` argument (or did you mean `@staticmethod`?)

Reply:
changed to `@staticmethod`

Reply:
renaming to `def static_method` might make the test clearer.

Reply:
Heck, you might even want to use two completely separate classes and tests here.

Reply:
good idea